server/src/test/java/org/candlepin/controller/ContentAccessManagerDBTest.java

```diff
@@ -0,0 +1,185 @@
+
+    private Owner createSCAOwner() {
+        Owner owner = this.createOwner();
+
+        owner.setContentAccessModeList(ENTITLEMENT_MODE + ", " + ORG_ENVIRONMENT_MODE);
+        owner.setContentAccessMode(ORG_ENVIRONMENT_MODE);
+
+        return this.ownerCurator.merge(owner);
+    }
+
+    private Consumer createV3Consumer(Owner owner, Environment environment) {
+        Consumer consumer = this.createConsumer(owner);
+
+        consumer.setFact("system.certificate_version", "3.0");
+        consumer.setEnvironment(environment);
+
+        return this.consumerCurator.merge(consumer);
+    }
+
+
+    private void scaCertGenerationTest(Consumer consumer) throws Exception {
+        ContentAccessManager manager = this.createManager();
+
+        ContentAccessCertificate cert = manager.getCertificate(consumer);
+
+        assertNotNull(cert);
+        assertNotNull(consumer.getContentAccessCert());
+    }
+
+    @Test
+    public void testGetCertificate() throws Exception {
+        Owner owner = this.createSCAOwner();
+        Consumer consumer = this.createV3Consumer(owner, null);
+
+        this.scaCertGenerationTest(consumer);
+    }
+
+    @Test
+    public void testGetCertificateWithEnvironment() throws Exception {
+        Owner owner = this.createSCAOwner();
+        Environment environment = this.createEnvironment(owner);
+        Consumer consumer = this.createV3Consumer(owner, environment);
+
+        this.scaCertGenerationTest(consumer);
+    }
+
+    private void regenerateExpiredSCACertTest(Consumer consumer) throws Exception {
+        ContentAccessManager manager = this.createManager();
+
+        ContentAccessCertificate cert = manager.getCertificate(consumer);
+        String oldCert = cert.getCert();
+
+        assertNotNull(cert);
+        assertNotNull(consumer.getContentAccessCert());
+
+        // Expire the cert, then run through the cycle again.
+        ContentAccessCertificate consumerCert = consumer.getContentAccessCert();
+        CertificateSerial oldSerial = consumerCert.getSerial();
+
+        oldSerial.setExpiration(Util.yesterday());
+        oldSerial = this.certSerialCurator.merge(oldSerial);
+
+        // This should trigger a new cert to be generated which should end up with
+        // a different serial than the one we have above.
+        ContentAccessCertificate newCert = manager.getCertificate(consumer);
+
+        assertNotNull(newCert);
+        assertNotNull(consumer.getContentAccessCert());
+
+        assertNotEquals(newCert.getCert(), oldCert);
+        assertNotEquals(newCert.getSerial().getId(), oldSerial.getId());
+    }
+
+    @Test
+    public void testGetCertificateRegeneratesExpiredCerts() throws Exception {
+        Owner owner = this.createSCAOwner();
+        Consumer consumer = this.createV3Consumer(owner, null);
+
+        this.regenerateExpiredSCACertTest(consumer);
+    }
+
+    @Test
+    public void testGetCertificateRegeneratesExpiredCertsWithEnvironment() throws Exception {
```

Comment:
The tests in this class cover a good amount of use cases, e.g. new SCA cert creation with/without environment, expired SCA cert regeneration with/without environment, but there's another area missing: testing that an existing (non-expired) SCA cert has only it's content payload updated (not the x509/serial) with new content (could be a unit test here, or could be a spec test; up to you)

Reply:
Done. Added spec tests.